### PR TITLE
Make Meditor it's own routable "page"

### LINKED ISSUE #864: Back button in metadata behaves somewhat unexpectedly
Go to any DLP, then click the "metadata" button. Now click the browser back button, and you will arrive at whatever page you were on *before* the DLP. This is because the metadata button doesn't create a history entry, even though it looks like a new page.

We can't simply create that history entry, because then the back button would go back to the DLP, discarding all in-flight changes (users are supposed to click the "save" or "back" icons to get out of the meditor).

Perhaps we can create that history entry and also put an event listener on the page close event to pop up the warning message about unsaved work?

### PR DESCRIPTION
**Pull request by mvandenburgh**
Right now the Meditor isn't a routable view, it's just an overlay on top of the DLP. This means we get weird behavior like the back button not functioning as expected. 

This also has a side effect of fixing #954.
Fixes #864.

### CODE REVIEW

**Review by waxlamp — COMMENTED** [review]
Nice work on this, killing two birds with one stone.

One problem I just noticed (I thought it was introduced by this PR, but it exists in production right now too): if you make edits to metadata, it is possible to click on the browser back button and lose all those changes. I'll file an issue to track this.

Otherwise, this looks good to me but I'm leaving this review as a comment because I'm not fully qualified to vet the code itself; go with Daniel's review on this.

File: src/views/MetadataView/Meditor.vue [hunk 1]
```diff
@@ -309,6 +299,17 @@ export default defineComponent({
 
     const TransactionTracker = new MeditorTransactionTracker(editorInterface);
 
+    const closeEditor = () => {
```
Comment (dchiquito) [inline src/views/MetadataView/Meditor.vue:302]: This looks like it closes without prompting if the user has unsaved work. Am I missing something?

File: src/views/DandisetLandingView/DandisetLandingView.vue [hunk 2]
```diff
@@ -153,8 +132,6 @@ export default defineComponent({
       schema,
       user,
       userCanModifyDandiset,
-      edit,
-      readonly,
```
[the remark this answers is not in the document]
Comment (mvandenburgh) [reply src/views/DandisetLandingView/DandisetLandingView.vue:157]: `readonly` wasn't used anywhere. `edit` was a conditional that controlled whether or not the meditor was displayed and was set when the `edit` event was emitted.

File: src/views/MetadataView/MetadataView.vue [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      required: true,
+    },
+  },
+  setup(props) {
+    const currentDandiset = computed(() => store.state.dandiset.dandiset);
+    const schema = computed(() => store.state.dandiset.schema);
+    const meta = computed(() => (currentDandiset.value ? currentDandiset.value.metadata : {}));
+    const userCanModifyDandiset = computed(() => store.getters.dandiset.userCanModifyDandiset);
+
+    if (!currentDandiset.value) {
```
[the remark this answers is not in the document]
Comment (mvandenburgh) [reply src/views/MetadataView/MetadataView.vue]: Good catch, that's exactly what happens. There seems to be some issues with reactivity in VJSF that prevent it from rerendering the form after a URL change, so I had to add the `renderMeditor` ref as a workaround.

File: src/views/MetadataView/Meditor.vue (hunk 1 above)
Comment (mvandenburgh) [reply src/views/MetadataView/Meditor.vue:302]: `closeEditor()` is called after the check for unsaved work happens. 
First `exitMeditor()` is called, which checks whether or not there is unsaved work. If there isn't, it calls `closeEditor()`. If there is, the unsaved changes popup comes up, and clicking "yes" on that will call `closeEditor()`.

**Review by dchiquito — APPROVED** [review]

**Pull request merged**